File: src/main/java/com/github/unchama/seichiassist/effect/breaking/RoundedTask.kt
package com.github.unchama.seichiassist.effect.breaking

import org.bukkit.scheduler.BukkitRunnable

abstract class RoundedTask : BukkitRunnable() {
  private var round = 0

  abstract fun firstAction()

  abstract fun secondAction()

  fun otherwiseAction() {
    cancel()
  }

  override fun run() {
    round++
    when (round) {
      1 -> firstAction()
      2 -> secondAction()
      else -> otherwiseAction()
    }
  }
}

File: src/main/java/com/github/unchama/seichiassist/effect/breaking/RoundedTask.scala
package com.github.unchama.seichiassist.effect.breaking

import org.bukkit.scheduler.BukkitRunnable
abstract class RoundedTask  extends  BukkitRunnable() {
  private var round = 0

  abstract def firstAction()

  abstract def secondAction()

  def otherwiseAction() {
    cancel()
  }

  override def run() {
    round += 1
      round match {
      case 1 => firstAction()
      case 2 => secondAction()
      case _ => otherwiseAction()
    }
  }
}
